### PR TITLE
Remove a workaround in the update-openapigen.sh to run 'make test' on macOS

### PR DESCRIPTION
**Pull request by shotarok**
**What this PR does / why we need it**:

* Modify sed options in the update-openapigen.sh to run 'make test' on macOS
* I failed to run `make test` on macOS (v10.13.6, High Sierra) with the following error message:

```
$ make test
go generate ./pkg/... ./cmd/...
hack/update-codegen.sh
Generating deepcopy funcs
Generating clientset for serving:v1alpha2 at github.com/kubeflow/kfserving/pkg/client/clientset
Generating listers for serving:v1alpha2 at github.com/kubeflow/kfserving/pkg/client/listers
Generating informers for serving:v1alpha2 at github.com/kubeflow/kfserving/pkg/client/informers
hack/update-openapigen.sh
sed: 1: "pkg/apis/serving/v1alph ...": extra characters at the end of p command
make: *** [generate] Error 1
```

**Special notes for your reviewer**:

1. I'd appreciate checking if this change would work with GNU sed

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/365)

### CODE REVIEW

**Comment by yuzisun** [comment]
@shotarok haha, I am just about to send a PR for this, we can totally get rid of this walkaround :).

File: hack/update-openapigen.sh [hunk 1]
```diff
@@ -26,8 +26,8 @@ fi
 go run vendor/k8s.io/code-generator/cmd/openapi-gen/main.go --input-dirs github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha2,knative.dev/pkg/apis --output-package github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha2/ --go-header-file hack/boilerplate.go.txt
 
 # Workaroud error "spec redeclared in this block" in unit test.
-sed -i 's%spec "github.com/go-openapi/spec"%openapispec "github.com/go-openapi/spec"%g' pkg/apis/serving/v1alpha2/openapi_generated.go
-sed -i 's/spec\./openapispec\./g' pkg/apis/serving/v1alpha2/openapi_generated.go
+sed -i '' -e 's%spec "github.com/go-openapi/spec"%openapispec "github.com/go-openapi/spec"%g' pkg/apis/serving/v1alpha2/openapi_generated.go
+sed -i '' -e 's/spec\./openapispec\./g' pkg/apis/serving/v1alpha2/openapi_generated.go
```
Comment (yuzisun) [inline hack/update-openapigen.sh]: you can delete this walkaround if you move these variables(https://github.com/kubeflow/kfserving/blob/master/pkg/apis/serving/v1alpha2/kfservice_framework_tensorrt_test.go#L41)  inside the function, it should not be declared at package level which causes conflicts.

Comment (shotarok) [reply hack/update-openapigen.sh]: Thanks for your heads up! I confirm the fix passes all unit tests on my local machine. I'll push the commit and rename this PR.

**Comment by shotarok** [comment]
[MEMO] I confirmed `make test` works on my macOS.

<details>
<summary>operation logs</summary>

```
➜  kfserving git:(sed-for-macos) ✗ git reset HEAD --hard
HEAD is now at ec24b6a2 Remove a workaround by moving a conflicted variable into a function
➜  kfserving git:(sed-for-macos) ✗ make test  
go generate ./pkg/... ./cmd/...
hack/update-codegen.sh
Generating deepcopy funcs
Generating clientset for serving:v1alpha2 at github.com/kubeflow/kfserving/pkg/client/clientset
Generating listers for serving:v1alpha2 at github.com/kubeflow/kfserving/pkg/client/listers
Generating informers for serving:v1alpha2 at github.com/kubeflow/kfserving/pkg/client/informers
hack/update-openapigen.sh
go fmt ./pkg/... ./cmd/...
go vet ./pkg/... ./cmd/...
hack/verify-golint.sh
Congratulations!  All Go source files have been linted.
go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go crd --output-dir=config/default/crds

[debug] name="Ready"
[debug] type="string"
[debug] JSONPath=".status.conditions[?(@.type=='Ready')].status"
[debug] name="URL"
[debug] type="string"
[debug] JSONPath=".status.url"
[debug] name="Default Traffic"
[debug] type="integer"
[debug] JSONPath=".status.default.traffic"
[debug] name="Canary Traffic"
[debug] type="integer"
[debug] JSONPath=".status.canary.traffic"
[debug] name="Age"
[debug] type="date"
[debug] JSONPath=".metadata.creationTimestamp"CRD files generated, files can be found under path config/default/crds.
kustomize build config/default/crds -o config/default/crds/serving_v1alpha2_kfservice.yaml
go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go rbac --output-dir=config/default/rbac
RBAC manifests generated under 'config/default/rbac' directory
go test ./pkg/... ./cmd/... -coverprofile cover.out
?   	github.com/kubeflow/kfserving/pkg/apis	[no test files]
?   	github.com/kubeflow/kfserving/pkg/apis/serving	[no test files]
ok  	github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha2	8.692s	coverage: 35.5% of statements
?   	github.com/kubeflow/kfserving/pkg/client/clientset/versioned	[no test files]
?   	github.com/kubeflow/kfserving/pkg/client/clientset/versioned/fake	[no test files]
?   	github.com/kubeflow/kfserving/pkg/client/clientset/versioned/scheme	[no test files]
?   	github.com/kubeflow/kfserving/pkg/client/clientset/versioned/typed/serving/v1alpha2	[no test files]
?   	github.com/kubeflow/kfserving/pkg/client/clientset/versioned/typed/serving/v1alpha2/fake	[no test files]
?   	github.com/kubeflow/kfserving/pkg/client/informers/externalversions	[no test files]
?   	github.com/kubeflow/kfserving/pkg/client/informers/externalversions/internalinterfaces	[no test files]
?   	github.com/kubeflow/kfserving/pkg/client/informers/externalversions/serving	[no test files]
?   	github.com/kubeflow/kfserving/pkg/client/informers/externalversions/serving/v1alpha2	[no test files]
?   	github.com/kubeflow/kfserving/pkg/client/listers/serving/v1alpha2	[no test files]
?   	github.com/kubeflow/kfserving/pkg/constants	[no test files]
?   	github.com/kubeflow/kfserving/pkg/controller	[no test files]
ok  	github.com/kubeflow/kfserving/pkg/controller/kfservice	9.165s	coverage: 61.7% of statements
ok  	github.com/kubeflow/kfserving/pkg/controller/kfservice/reconcilers/knative	7.144s	coverage: 68.3% of statements
ok  	github.com/kubeflow/kfserving/pkg/controller/kfservice/resources/credentials	6.935s	coverage: 83.3% of statements
ok  	github.com/kubeflow/kfserving/pkg/controller/kfservice/resources/credentials/azure	0.095s	coverage: 100.0% of statements
ok  	github.com/kubeflow/kfserving/pkg/controller/kfservice/resources/credentials/gcs	0.053s	coverage: 100.0% of statements
ok  	github.com/kubeflow/kfserving/pkg/controller/kfservice/resources/credentials/s3	0.056s	coverage: 96.3% of statements
ok  	github.com/kubeflow/kfserving/pkg/controller/kfservice/resources/knative	7.616s	coverage: 63.7% of statements
ok  	github.com/kubeflow/kfserving/pkg/executor	0.054s	coverage: 56.2% of statements
?   	github.com/kubeflow/kfserving/pkg/testing	[no test files]
ok  	github.com/kubeflow/kfserving/pkg/utils	0.276s	coverage: 100.0% of statements
?   	github.com/kubeflow/kfserving/pkg/webhook	[no test files]
ok  	github.com/kubeflow/kfserving/pkg/webhook/admission/deployment	7.390s	coverage: 61.0% of statements
?   	github.com/kubeflow/kfserving/pkg/webhook/admission/kfservice	[no test files]
?   	github.com/kubeflow/kfserving/pkg/webhook/third_party	[no test files]
?   	github.com/kubeflow/kfserving/cmd/executor	[no test files]
?   	github.com/kubeflow/kfserving/cmd/manager	[no test files]
?   	github.com/kubeflow/kfserving/cmd/spec-gen	[no test files]
```

</details>

File: pkg/apis/serving/v1alpha2/kfservice_framework_tensorrt_test.go [hunk 2]
```diff
@@ -52,6 +46,11 @@ var config = FrameworksConfig{
 
 func TestCreateModelServingContainer(t *testing.T) {
 
+	var spec = TensorRTSpec{
```
Comment (yuzisun) [inline pkg/apis/serving/v1alpha2/kfservice_framework_tensorrt_test.go]: while you are on this, can you help move `requestedResource` and `config` inside test function as well? Thanks!

Comment (shotarok) [reply pkg/apis/serving/v1alpha2/kfservice_framework_tensorrt_test.go]: c213948 I moved them too

**Comment by yuzisun** [comment]
Awesome! Thanks for the fix @shotarok !

/lgtm
/approve

**Pull request merged**